ports/servoshell/egl/host_trait.rs

```diff
@@ -3,18 +3,23 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 use servo::webrender_api::units::DeviceIntRect;
-use servo::{InputMethodType, LoadStatus, MediaSessionPlaybackState, PromptResult};
+use servo::{
+    AlertResponse, ConfirmResponse, InputMethodType, LoadStatus, MediaSessionPlaybackState,
+    PromptResponse,
+};
 
-/// Callbacks. Implemented by embedder. Called by Servo.
+/// Callbacks implemented by embedder. Called by our RunningAppState, generally on behalf of Servo.
 pub trait HostTrait {
-    /// Show alert.
-    fn prompt_alert(&self, msg: String, trusted: bool);
-    /// Ask Yes/No question.
-    fn prompt_yes_no(&self, msg: String, trusted: bool) -> PromptResult;
-    /// Ask Ok/Cancel question.
-    fn prompt_ok_cancel(&self, msg: String, trusted: bool) -> PromptResult;
-    /// Ask for string
-    fn prompt_input(&self, msg: String, default: String, trusted: bool) -> Option<String>;
+    /// Show a trusted alert from Servo or the embedder.
+    fn show_trusted_alert(&self, message: String);
+    /// Show a trusted Yes/No prompt from Servo or the embedder.
+    fn show_trusted_yes_no_dialog(&self, message: String) -> YesNoResponse;
```

Comment:
Could we use this opportunity to pass the `request` object through to the embedder, so that it can be handled asynchronously?

Reply:
yeah, that’s the plan. this patch was in preparation for finishing the new delegate request pattern, which i’ll do in separate patches :)